drivers/i2c/i2c_renesas_rx_riic.c

```diff
@@ -0,0 +1,384 @@
+	data->rdp_info.p_data2nd = buf2 ? buf2 : FIT_NO_PTR;
+	data->rdp_info.p_slv_adr = (uint8_t *)addr;
+}
+
+static int run_rx_transfer(const struct device *dev, struct i2c_msg *msgs, uint8_t num_msgs,
+			   uint16_t addr, bool async)
+{
+	struct i2c_rx_data *data = (struct i2c_rx_data *const)dev->data;
+	riic_return_t rdp_ret;
+
+	k_mutex_lock(&data->bus_lock, K_FOREVER);
+	k_sem_reset(&data->bus_sync);
+
+	if (addr == 0x00) {
+		/* Enter transmission pattern 4 */
+		LOG_DBG("RDP RX I2C master transmit pattern 4\n");
+		setup_rdp_info(data, NULL, 0, NULL, 0, NULL);
+		rdp_ret = R_RIIC_MasterSend(&data->rdp_info);
+		goto TRANSFER_BLOCKING;
+	}
+
+	if (num_msgs == 1) {
+		if (msgs[0].flags & I2C_MSG_READ) {
+			/* Enter master reception pattern 1 */
+			LOG_DBG("RDP RX I2C master reception pattern 1\n");
+			setup_rdp_info(data, NULL, 0, msgs[0].buf, msgs[0].len, &addr);
+			rdp_ret = R_RIIC_MasterReceive(&data->rdp_info);
+			goto TRANSFER_BLOCKING;
+		} else {
+			/* Enter master transmission pattern 2/3 */
+			LOG_DBG("RDP RX I2C master transmit pattern 2/3\n");
+			setup_rdp_info(data, NULL, 0, msgs[0].len ? msgs[0].buf : NULL, msgs[0].len,
+				       &addr);
+			rdp_ret = R_RIIC_MasterSend(&data->rdp_info);
+			goto TRANSFER_BLOCKING;
+		}
+	} else if (num_msgs == 2) {
+		if (msgs[0].flags & I2C_MSG_READ) {
+			goto UNSUPPORT_PATTERN;
+		}
+
+		if (msgs[1].flags & I2C_MSG_READ) {
+			/* Enter master reception pattern 2 */
+			LOG_DBG("RDP RX I2C master reception pattern 2\n");
+			setup_rdp_info(data, msgs[0].buf, msgs[0].len, msgs[1].buf, msgs[1].len,
+				       &addr);
+			rdp_ret = R_RIIC_MasterReceive(&data->rdp_info);
+			goto TRANSFER_BLOCKING;
+		} else {
+			/* Enter master transmission pattern 1 */
+			LOG_DBG("RDP RX I2C master transmit pattern 1\n");
+			setup_rdp_info(data, msgs[0].buf, msgs[0].len, msgs[1].buf, msgs[1].len,
+				       &addr);
+			rdp_ret = R_RIIC_MasterSend(&data->rdp_info);
+			goto TRANSFER_BLOCKING;
+		}
+	}
+
+UNSUPPORT_PATTERN:
+	/* Unsupport pattern, emit each fragment as a distinct transaction  */
+	LOG_DBG("%s: \"Not a generic pattern ...\" !\n", __func__);
+	for (uint8_t i = 0; i < num_msgs; i++) {
+		if (msgs[i].flags & I2C_MSG_READ) {
+			/* Enter master reception pattern 1 */
+			LOG_DBG("RDP RX I2C master reception pattern 1\n");
+			setup_rdp_info(data, NULL, 0, msgs[i].buf, msgs[i].len, &addr);
+			rdp_ret = R_RIIC_MasterReceive(&data->rdp_info);
+		} else {
+			/* Enter master transmission pattern 2 */
+			LOG_DBG("RDP RX I2C master transmit pattern 2/3\n");
+			setup_rdp_info(data, NULL, 0, (msgs[i].len) ? msgs[i].buf : NULL,
+				       msgs[i].len, &addr);
+			rdp_ret = R_RIIC_MasterSend(&data->rdp_info);
+		}
+		if (i == num_msgs - 1) {
+			data->skip_callback = false; /* Invoke callback in the last message */
+		}
+		if (!rdp_ret) {
+			k_sem_take(&data->bus_sync, K_FOREVER);
+			k_sem_reset(&data->bus_sync);
+		} else {
+			break;
+		}
+	}
+	goto TRANSFER_EXIT;
```

Comment:
no need to all caps labels, zephyr code typically uses lower case labels for goto